libs/zedUpload/awsutil/s3sync.go

```diff
@@ -23,22 +23,12 @@ import (
 	"github.com/lf-edge/eve/libs/zedUpload/types"
 )
 
-// stats update
-type UpdateStats struct {
-	Name      string                // always the remote key
-	Size      int64                 // complete size to upload/download
-	Asize     int64                 // current size uploaded/downloaded
-	List      []string              //list of images at given path
-	DoneParts types.DownloadedParts //downloaded parts
-}
-
-type NotifChan chan UpdateStats
-
 // CustomReader contains the details of Chunks being downloaded
 type CustomReader struct {
 	fp        *os.File
-	upSize    UpdateStats
-	prgNotify NotifChan
+	name      string
```

Comment:
You have a bunch of new and updated fields in different structs called "name" and "Name" without any comment explaining what they are. Are they only for better logging? Some comparison? Please document in comments.

Reply:
Will do. They were mostly buried inside of each of the type-specific  `UpdateStats`, which wasn't the best place for them, so I extracted them out. I will add comments.

Reply:
To be honest, I am not sure what they are used for. They appear in 4 places, each of `customReader` and `customWriter` on AWS s3sync and Google Cloud Storage. In all of those cases, it passed the name on to the update stats as follows:

```go
	creader := &CustomReader{
 		fp:        file,
 		upSize:    UpdateStats{Size: fileInfo.Size(), Name: bkey},
 		prgNotify: prgNotify,
 	}
```

Transitioning that to a standard `UpdateStats`, left the `Name` outside, so I placed it as a separate field:

```go
	creader := &CustomReader{
 		fp:        file,
 		upSize:    types.UpdateStats{Size: fileInfo.Size()},
 		name:      bkey,
 		prgNotify: prgNotify,
 	}
```

But I can find nowhere that it is consumed. If you have no objection, I am fine just removing it entirely. As long as CI doesn't complain, and we cannot find it anywhere, any reason why not?

Reply:
I don't know if CI currently covers all of the code paths for all of the zedUpload transports. I don't have any issue with leaving them in now that you explained you just moved them up to the common struct.

Reply:
Seems this field (name) not used by customReader or customWriter, and do not propagated anywhere, so I think we can remove it.